specs/merge/beacon-chain.md

```diff
@@ -59,148 +58,138 @@ We define the following Python custom types for type hinting and readability:
+    receipt_root: Bytes32  # 'receipts root' in the yellow paper
+    block_number: uint64  # 'number' in the yellow paper
     gas_limit: uint64
     gas_used: uint64
     timestamp: uint64
-    receipt_root: Bytes32
-    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
-    transactions: List[OpaqueTransaction, MAX_EXECUTION_TRANSACTIONS]
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
+    transactions: List[Transaction, MAX_TRANSACTIONS_PER_PAYLOAD]
 ```
 
 #### `ExecutionPayloadHeader`
 
-The execution payload header included in a `BeaconState`.
-
-*Note:* Holds execution payload data without transaction bodies.
-
 ```python
 class ExecutionPayloadHeader(Container):
-    block_hash: Hash32  # Hash of execution block
+    # Execution block header fields
     parent_hash: Hash32
     coinbase: Bytes20
     state_root: Bytes32
-    number: uint64
+    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    receipt_root: Bytes32
+    block_number: uint64
     gas_limit: uint64
     gas_used: uint64
     timestamp: uint64
-    receipt_root: Bytes32
-    logs_bloom: ByteVector[BYTES_PER_LOGS_BLOOM]
+    # Extra payload fields
+    block_hash: Hash32  # Hash of execution block
     transactions_root: Root
 ```
 
-## Protocols
-
-### `ExecutionEngine`
-
-The `ExecutionEngine` protocol separates the consensus and execution sub-systems.
-The consensus implementation references an instance of this sub-system with `EXECUTION_ENGINE`. 
-
-The following methods are added to the `ExecutionEngine` protocol for use in the state transition:
-
-#### `new_block`
+## Helper functions
 
-Verifies the given `execution_payload` with respect to execution state transition, and persists changes if valid.
+### Predicates
 
-The body of this function is implementation dependent.
-The Consensus API may be used to implement this with an external execution engine.
+#### `is_merge_complete`
 
 ```python
-def new_block(self: ExecutionEngine, execution_payload: ExecutionPayload) -> bool:
-    """
-    Returns True if the ``execution_payload`` was verified and processed successfully, False otherwise. 
-    """
-    ...
+def is_merge_complete(state: BeaconState) -> bool:
+    return state.latest_execution_payload_header != ExecutionPayloadHeader()
 ```
 
-## Helper functions
+#### `is_merge_block`
 
-### Misc
+```python
+def is_merge_block(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return not is_merge_complete(state) and body.execution_payload != ExecutionPayload()
+```
 
 #### `is_execution_enabled`
 
 ```python
-def is_execution_enabled(state: BeaconState, block: BeaconBlock) -> bool:
-    return is_transition_completed(state) or is_transition_block(state, block)
+def is_execution_enabled(state: BeaconState, body: BeaconBlockBody) -> bool:
+    return is_merge_block(state, body) or is_merge_complete(state)
 ```
 
-#### `is_transition_completed`
+### Misc
 
-```python
-def is_transition_completed(state: BeaconState) -> bool:
```

Comment:
This function is used in [fork-choice.md](https://github.com/ethereum/eth2.0-specs/blob/dev/specs/merge/fork-choice.md#on_block).

Reply:
Understood!

As a heads up this function has not been removed, merely renamed :) Making the naming consistent outside of `beacon-chain.md` (e.g. in `fork-choice.md`) is on my todo list (hence the "draft" status of this PR). I'll resolve this point for now—feel free to reopen if you disagree.